File: netflix-prize-graph-example/python/average_rating.py
import sys

from netflix_dal import get_average_rating_of_movie

if len(sys.argv) != 2:
    print('Usage: average_rating <movie_id>')
    exit(1)

movie_id = sys.argv[1]
try:
    result = get_average_rating_of_movie(int(movie_id))

    if result is None:
        print(f'There is no movie with ID {movie_id}.')
        exit(0)

    print(f'The average rating of movie ID {movie_id} is {result}.')
except ValueError:
    print(f'Sorry, something went wrong. Please ensure that “{movie_id}” is a valid movie ID.')

File: netflix-prize-graph-example/nodejs/average-rating.mjs
import { getAverageRatingOfMovie } from './netflix-dal.mjs'

if (process.argv.length !== 3) {
  console.log('Usage: average-rating <movieId>')
  process.exit(1)
}

const movieId = process.argv[2]
try {
  const result = await getAverageRatingOfMovie(movieId)
  if (!result) {
    console.log(`There is no movie with ID ${movieId}.`)
    process.exit(0)
  }

  console.log(`The average rating of movie ID ${movieId} is ${result}.`)
} catch (error) {
  console.error(`Sorry, something went wrong. Please ensure that “${movieId}” is a valid movie ID.`)
}

// Explicitly exit sync since we are asynchronous at this point.
process.exit(0)
